### DISCUSSION TITLE
File deletion

### DISCUSSION THREAD

**Discussion by pedalracer** [sumatrapdfreader/sumatrapdf#4031] [Ideas]
I really like to have the menu option to directly delete a pdf into trash bin.
Since 25 years I have to go back into the explorer in order to delete a pdf I have read in a pdf viewer.
Normal workflow, you know, is find a pdf online etc, download it, find it in explorer, open it in pdf viewer, read it, close it, got to explorer find it in order to delete it, if you decided that its worth to go to trash bin.

What a hassle just to write this down. I hate it :) So ! Please be the first to add option in pdf viewer to delete open pdf into trash bin.

Thanks

**Comment by GitHubRulesOK** [comment]
the reason you cant delete a file whilst it is open for viewing is it would most likely crash an application. Thus from the first days of bugs crawling into computers the files are locked from deletion whilst viewing. SumatraPDF makes a concession ONLY for small PDFs (under 32 MB so they can be LaTeX edited whist pre- viewing) but the system should also be file locking against destructive activities.

What I have repeatedly shown is how to save the current file to a deletion list then on closing all applications like viewers or previewers or antivirus etc. the file list of deletions be run.

Example from 2022 https://github.com/sumatrapdfreader/sumatrapdf/discussions/2368#discussioncomment-2232367 so just tested it now on a couple of files and it put this on the desktop for later

**DeletionList.cmd**
```
del "C:\Users\lez\Downloads\Apps\PDF\Data\jpgTOpdf (2)\31mpF.pdf" 
del C:\Users\lez\Downloads\9ffb21ca-717e-c712-65de-91880bccebdc.jpg 
```

**Comment by pedalracer** [comment]
I use the marvelous program called FileJuggler. I think I will get it done in a way you suggested. 
What works is, to rename PDF in Sumatra (F2). Possible to do with open pdf! Then have tool to look for specific filename (del_anypdf.pdf) and move it to specific folder or delete it.
Thanks
